runtime/notification.go

```diff
@@ -0,0 +1,71 @@
+package runtime
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+type Notification struct {
+	GroupID   string      `json:"group_id"`
+	EventType string      `json:"event_type"`
+	Event     interface{} `json:"event"`
+}
+
+type StageEvent struct {
+	Name   string `json:"name"`
+	Action string `json:"action"` // entry or exit
+}
+
+type OutcomeEvent struct {
+	Outcome string `json:"outcome"` // ok
+}
+
+func NewStageNotification(groupId, name, action string) *Notification {
+	return &Notification{
+		GroupID:   groupId,
+		EventType: "stage",
+		Event: &StageEvent{
+			Name:   name,
+			Action: action,
+		},
+	}
+}
+
+func NewOutcomeOKNotification(groupId string) *Notification {
+	return &Notification{
+		GroupID:   groupId,
+		EventType: "outcome",
+		Event: &OutcomeEvent{
+			Outcome: "ok",
+		},
+	}
+}
+
+func (n *Notification) ParseEvent() error {
+	if n.EventType == "stage" {
+		r := &StageEvent{}
+		err := mapstructure.Decode(n.Event, r)
+		if err != nil {
+			return err
+		}
+
+		n.Event = r
+
+		return nil
+	}
+
+	if n.EventType == "outcome" {
+		r := &OutcomeEvent{}
+		err := mapstructure.Decode(n.Event, r)
```

Comment:
I don't think you need mapstructure for this?

Reply:
Also, I don't see event types being user-defined nor very diverse in data model, so I don't think the polymorphism is necessary here.

Reply:
We have one `channel` for `notification` (or events, whatever you want to call them). When we consume them from the `daemon`, we don't know what type we are consuming. Currently we have 2 types - OutcomeEvent and StageEvent - how is the daemon going to know what notification it gets, when we use a single channel, without some kind of polymorphism?

Reply:
Right. We should be able to achieve this with `json.Message` from `encoding/json`!

Reply:
If you're referring to `json.RawMessage`, I am not sure the solution is much nicer to what we have here... could you point me to an example you like where we marshal/unmarshal a struct with an interface{} field in it?

Reply:
Hmm, what I'm saying is that we don't need a 3rd party lib because the functionality we're relying on here is available in the standard `encoding/json` package. Unless I'm missing something...

Reply:
We are already using `mapstructure` elsewhere in Testground, so this is not the first place we include it.

Using `encoding/json` is totally workable, but we already solve that same problem on Testground with `mapstructure`.